x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -540,6 +546,54 @@ public void testGetApiKeysOwnedByCurrentAuthenticatedUser() throws InterruptedEx
             response, userWithManageApiKeyRoleApiKeys.stream().map(o -> o.getId()).collect(Collectors.toSet()), null);
     }
 
+    public void testGetApiKeysOwnedByRunAsUserWhenOwnerIsTrue() throws ExecutionException, InterruptedException {
+        createUserWithRunAsRole();
+        int noOfSuperuserApiKeys = randomIntBetween(3, 5);
+        int noOfApiKeysForUserWithManageApiKeyRole = randomIntBetween(3, 5);
+        createApiKeys(noOfSuperuserApiKeys, null);
```

Comment:
Is this line needed?

Reply:
Not strictly needed. I copied it from the existing test and my understanding is that it adds some noise during test setup. After a series of API key operations, the verification step makes sure the operations touch only designated keys and *not* the noise. I'll just leave them as is since it has some (marginal) value.